### PR TITLE
fix(executor): More logs for PNS sidecar termination. #5627

### PR DESCRIPTION
**Pull request by alexec**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -129,12 +129,12 @@ define protoc
      perl -i -pe 's|argoproj/argo-workflows/|argoproj/argo-workflows/v3/|g' `echo "$(1)" | sed 's/proto/pb.go/g'`
 
 endef
-# docker_build,image_name
+# docker_build,target,image_name
```
Comment (alexec) [inline Makefile:132]: fixes a bug in build where locally built argoexec did not get imported into k3s

**Comment by alexec** [comment]
Failed PNS build:

https://github.com/argoproj/argo-workflows/pull/5683/checks?check_run_id=2355710905

```
34m          logs | [msidecar-injected-8l77l wait time="2021-04-15T18:38:18.225Z" level=info msg="guessing container \"main\" PID is 28"
[34m          logs | [msidecar-injected-8l77l wait time="2021-04-15T18:38:18.225Z" level=info msg="Waiting for \"main\" pid 28 to complete"
[34m          logs | [msidecar-injected-8l77l wait time="2021-04-15T18:38:18.232Z" level=info msg="secured root for pid 28 root: sh"
```

**Comment by alexec** [comment]
@sarabala1979 please review.

File: workflow/executor/pns/pns.go [hunk 2]
```diff
@@ -157,32 +157,6 @@ func (p *PNSExecutor) Wait(ctx context.Context, containerNames []string) error {
 		time.Sleep(1 * time.Second)
 	}
 
-OUTER:
```
Comment (alexec) [inline workflow/executor/pns/pns.go:160]: I don't think it is worthwhile getting this to work as the performance benefit is much smaller than the bugginess.

**Review by sarabala1979 — APPROVED** [review]

File: workflow/executor/pns/pns.go [hunk 3]
```diff
@@ -331,12 +305,15 @@ func (p *PNSExecutor) secureRootFiles() error {
...
-			// the main container may have switched (e.g. gone from busybox to the user's container)
-			if prevInfo, ok := p.pidFileHandles[pid]; ok {
-				_ = prevInfo.Close()
+			if p.pidFileHandles[pid] != fs {
+				// the main container may have switched (e.g. gone from busybox to the user's container)
+				if prevInfo, ok := p.pidFileHandles[pid]; ok {
+					_ = prevInfo.Close()
+				}
+				p.pidFileHandles[pid] = fs
+				log.Infof("ALEX secured root for pid %d root: %s (%q)", pid, proc.Executable(), fs.Name())
```
Comment (dinever) [inline workflow/executor/pns/pns.go:314]: Just curious. Is this part `ALEX ..` for debugging puporse?

Comment (alexec) [reply workflow/executor/pns/pns.go:314]: ops...

**Pull request merged**